frontend/public/components/deploy-image.tsx

```diff
@@ -48,9 +49,12 @@ const ImagePorts = ({ports, name}) => <React.Fragment>
 </React.Fragment>;
 
 export class DeployImage extends React.Component<DeployImageProps, DeployImageState> {
+
+  private environmentEditor: any;
+
   constructor (props) {
     super(props);
-
+    this.save = this.save.bind(this);
```

Comment:
This isn't needed. `save` is already bound to `this` since it's declared using `=>` below.